File: Cloudy.CMS.UI/wwwroot/FormSupport/Controls/text-control.js
import FieldControl from '../field-control.js';

class TextControl extends FieldControl {
    constructor(fieldModel, value, app, blade, originalValue) {
        var container = document.createElement('div');
        super(container);

        this.contentId = blade?.content?.id;
        this.contentTypeId = blade?.contentType?.id;
        this.changeTracker = app.changeTracker;
        this.path = fieldModel.descriptor.camelCaseId;
        this.name = this.changeTracker?.buildControlName(this.contentTypeId, this.contentId, fieldModel.descriptor.camelCaseId);
        this.backupValue = originalValue || value;
      
        var input = document.createElement('input');
        input.classList.add('cloudy-ui-form-input');

        if (fieldModel.descriptor.control.uiHint == 'password') {
            input.type = 'password';
        } else {
            input.type = 'text';
        }
        input.value = value || null;
        input.name = this.name;
        container.append(input);
        input.addEventListener('change', () => this.triggerChange(input.value || null));
        input.addEventListener('keyup', () => this.triggerChange(input.value || null));

        this.onSet(value => input.value = value || null);
    }
}

export default TextControl;

File: Cloudy.CMS.UI/FormSupport/Controls/TextControl.cs
using Cloudy.CMS.UI.FormSupport.ControlSupport;
using System;
using System.Collections.Generic;
using System.Text;

namespace Cloudy.CMS.UI.FormSupport.Controls
{
    [Control("text", "edit-content/form/controls/text.js")]
    [MapControlToType(typeof(double))]
    [MapControlToType(typeof(double?))]
    [MapControlToType(typeof(int))]
    [MapControlToType(typeof(int?))]
    [MapControlToType(typeof(long))]
    [MapControlToType(typeof(long?))]
    [MapControlToType(typeof(string))]
    [MapControlToUIHint("password")]
    public class TextControl
    {
    }
}
